File: c/puzzles/is_prime.c
#include <stdlib.h>
#include <stdio.h>
#include <math.h>

// gcc -Wall prime.c -lm

int main(int argc, char **argv) {
    if (argc == 1) {
        printf("Usage: %s number\n", argv[0]);
        exit(1);
    }

    short int n = atoi(argv[1]);

    if (n < 2 || n % 2 == 0) {
        printf("%d is not prime\n", n);
        exit(2);
    }

    short int i = 3;
    int p = (int) sqrt(n);

    for (; i <= p; ++i) {
        if (n % i == 0) {
            printf("%d is not prime\n", n);
            exit(0);
        }
    }

    printf("%d is prime\n", n);
    return 0;
}

File: python/puzzles/is_prime.py
from math import sqrt


def is_prime(n):
    if n < 2:
        return False

    for k in range(2, int(sqrt(n)) + 1):
        if n % k == 0:
            return False

    return True


if __name__ == "__main__":
    is_prime()

#[n for n in range(100) if is_prime(n)]
#{n*n: (1, n, n*n) for n in range(101) if is_prime(n)}

